### PR TITLE
docs: document relay false-negative RCA

### PR DESCRIPTION
**Pull request by seungpyoson**
## Summary

- Add a detailed root-cause analysis for the original relay approval false-negative investigation.
- Document provider-by-provider symptoms, route ladder behavior, PR #214 fixes, and remaining follow-up work.
- Add an evidence dossier with transcript, PR, code, test, route, Grok, Kimi, and issue #215 evidence plus confidence boundaries.

## Verification

- `git diff --check`
- `git diff --cached --check`

Related to #216 and #215.

### CODE REVIEW

File: docs/relay-false-negative-root-cause-analysis.md [hunk 1]
```diff
@@ -0,0 +1,900 @@
...
+This section is intentionally more literal than the RCA sections above. It
+records the hard evidence used to derive the conclusions, including verbatim
+transcript excerpts, PR text, code excerpts, test assertions, and confidence per
+claim.
+
+### Evidence Sources
+
+| ID | Source | What It Proves | Verification Method |
+| --- | --- | --- | --- |
+| S1 | `/Users/spson/.codex/sessions/2026/06/06/rollout-2026-06-06T02-22-22-019e98ce-ade3-79e3-8586-919135671a46.jsonl` | Original session symptom: provider split, policy denial class, source not sent for blocked providers | `jq` extraction of user/assistant messages matching relay/policy terms |
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply docs/relay-false-negative-root-cause-analysis.md]: Fixed in `7a047af` by replacing the absolute home path with `~/.codex/...` while preserving the exact session filename.

Verification:
- `rg -n "/Users/spson" docs/relay-false-negative-root-cause-analysis.md` returns no matches.
- `git diff --check` passes.
